### PR TITLE
update `--db-user` and `--db-name` docs

### LINKED ISSUE #32307: `--db-name` is no longer optional in `tsh db login`
Expected behavior:

Running `tsh db login --db-user=user my-database`

should login to the teleport database. 

This was the behaviour prior to upgrading to `14.0.0`. 

Current behavior:

```
tsh db login --db-user=user my-database
ERROR: please provide the database name using the --db-name flag, allowed database names for my-database: [*]
```

The documentation from `tsh db login --help` shows `--db-name` as optional still. 

```
 --db-name                  Optional database name to configure as default.
```

Bug details:
- Teleport version 14.0.0

### PR DESCRIPTION
**Pull request by GavinFrazar**
Closes https://github.com/gravitational/teleport/issues/32307

This PR clarifies when `--db-name` or `--db-user` are required/optional flags, and how defaults are selected first from an active cert or from the user's allow db user/names.

### CODE REVIEW

File: docs/pages/includes/db-user-name-flags.mdx [hunk 1]
```diff
@@ -0,0 +1,6 @@
+If `--db-user` or `--db-name` are required but not given, then default settings
```
[the remark this answers is not in the document]
Comment (GavinFrazar) [reply docs/pages/includes/db-user-name-flags.mdx:1]: We don't choose default from allowed db names if the db name isn't required for the protocol though

File: docs/pages/database-access/reference/cli.mdx [hunk 2]
```diff
@@ -295,8 +295,10 @@ $ tsh db login --db-user=postgres --db-name=postgres example
 
 | Flag | Description |
 | - | - |
-| `--db-user` | Optionally, set default database account name to connect as. |
-| `--db-name` | Optionally, set default database name to connect to. |
+| `--db-user` | The database user to log in as. |
+| `--db-name` | The database name to log in to. |
```
Comment (bl-nero) [inline docs/pages/database-access/reference/cli.mdx:299]: Nit: is it "log in to", or "log into"?

**Review by bl-nero — APPROVED** [review]

File: docs/pages/database-access/reference/cli.mdx (hunk 2 above)
Comment (GavinFrazar) [reply docs/pages/database-access/reference/cli.mdx:299]: login to
log into
log in to

I googled it when I made the changes too lol. Proper form is "log in to"

File: docs/pages/includes/db-user-name-flags.mdx (hunk 1 above)
[the remark this answers is not in the document]
Comment (zmb3) [reply docs/pages/includes/db-user-name-flags.mdx:1]: Hmm, "required but not given" is a bit of an oxymoron isn't it?

[the remark this answers is not in the document]
Comment (GavinFrazar) [reply docs/pages/includes/db-user-name-flags.mdx:1]: I guess we can just say "are required, then..."

**Comment by GavinFrazar** [comment]
@ptgott

File: docs/pages/database-access/reference/cli.mdx [hunk 3]
```diff
@@ -327,8 +329,10 @@ $ tsh db connect --db-user=alice --db-name=db example
 
 | Flag | Description |
 | - | - |
-| `--db-user` | Optionally, set database user name to connect as. |
-| `--db-name` | Optionally, set database name to connect to. |
+| `--db-user` | The database user to connect as. |
+| `--db-name` | The database name to connect to. |
```
Comment (zmb3) [inline docs/pages/database-access/reference/cli.mdx:333]: Let's use language in both places. I don't care if we say "to log in as" or "to connect as" but we should pick one and be consistent.

**Review by zmb3 — APPROVED** [review]

**Pull request merged**